### ISSUE TITLE
Question regarding wired sensors

### ISSUE THREAD

**Issue by garyhooper** [EHylands/homebridge-qolsys#31]
I'm upgrading from an old-school dumb alarm system, and strongly considering Qolsys IQ4.  All doors and windows are hardwired.

In addition to security functions, I would like for my Homebridge server to know when a window/door sensor changes state.  In this way, I want to be able "turn off the AC if the door is subsequently opened for longer than 5 minutes", or "send a message if windows are open and the temperature drops too low".  I know that Qolsys can communicate with certain AC systems, but Homekit/Homebridge is the center of my automation logic, and Qolsys does not integrate with my HVAC.

Can you confirm that your plug-in enables such functionality?   Is access to the Qolsys local, on WiFi, or remote through a Qolsys site or Alarm.com?  Any other salient tips for my intended use?

Thanks!

**Comment by EHylands** [comment]
> I'm upgrading from an old-school dumb alarm system, and strongly considering Qolsys IQ4.  All doors and windows are hardwired.
> 
> 
> 
> In addition to security functions, I would like for my Homebridge server to know when a window/door sensor changes state.  In this way, I want to be able "turn off the AC if the door is subsequently opened for longer than 5 minutes", or "send a message if windows are open and the temperature drops too low".  I know that Qolsys can communicate with certain AC systems, but Homekit/Homebridge is the center of my automation logic, and Qolsys does not integrate with my HVAC.
> 
> 
> 
> Can you confirm that your plug-in enables such functionality?   Is access to the Qolsys local, on WiFi, or remote through a Qolsys site or Alarm.com?  Any other salient tips for my intended use?
> 
> 
> 
> Thanks!

Hi,
First of all, access to the Qolsys panel is local through the fairly limited Control4 interface.  A remote subscription is not needed. Dealer code for your panel is required to enable this local integration.

I have not been experimenting with wired sensors myself, but from my understanding you would be using a iq hardwire translator. From a plugin perspective, these wired sensors will be exposed as regular wireless sensors without additional Homebridge configuration.

Each Qolsys sensor/zone gets exposed as a Homekit sensor and regular Homekit automations may be used.

I don't think you can achieve your particular needs with regular basic Homekit automations though. The other way around could be to convert automations to shortcuts and scripting or use homebridge delay dummy switches for complex patterns.

I hope it answers your questions :)

**Comment by garyhooper** [comment]
Hi,

To clarify, the limited Control 4 interface is a public API created by Control 4 to talk to the Qolsys?  Hence the limitation?

Yes, there is a wired-wireless bridge required, to expose each wired sensor to the Qolsys panel.  My goal is HomeKit arm/disarm, knowing the sensor state, and detecting whether the state changes.  This also enables, e.g., "When backdoor opens, turn on backdoor light."

 I was planning to use Homebridge delay switches to enable.  (Delays should really be native to Homekit.)

**Comment by EHylands** [comment]
Control4 wrote a driver in 2019 allowing direct connexion to supported Qolsys panels 
(Qolsys wrote a server running on port 12345 of their panels allowing interraction with that Control4 driver)

API/integration is officialy undocumented by Qolsys.
Basic functionality has been documented by people on Home Assistant forum.
Connexion is all local on your home network. All events on your panel will be pushed almost instantaneously to Homekit.

Homebridge plugin allows you to:
- Arm/Disarm partitions on your Qolsys panel through the Homekit security accessory and automations.
- Get notifications in Homekit when an alarm is triggered on your panel
- Get an almost instantaneous Homekit sensor state change for any sensors events on your Qolsys panel and trigger associated Homekit automations.

Anything related to zwave on Qolsys panel is not supported by the integration and this homebridge plugin.

To sum it up, you can achieve your goal with this plugin !

**Comment by garyhooper** [comment]
Thank!
